mitiq/lre/tests/test_multivariate_richardson.py

```diff
@@ -0,0 +1,239 @@
+    )
+
+    assert calculated_basis == expected_basis
+
+
+qreg1 = LineQubit.range(3)
+test_circuit1 = Circuit(
+    [ops.H.on_each(*qreg1)],
+    [ops.CNOT.on(qreg1[0], qreg1[1])],
+)
+test_circuit2 = Circuit(
+    [ops.H.on_each(*qreg1)],
+    [ops.CNOT.on(qreg1[0], qreg1[1])],
+    [ops.X.on(qreg1[2])],
+    [ops.TOFFOLI.on(*qreg1)],
+)
+
+
+@pytest.mark.parametrize(
+    "test_circ, test_degree, expected_matrix",
+    [
+        (
+            test_circuit1,
+            2,
+            np.array(
+                [
+                    [1.0, 1.0, 1.0, 1.0, 1.0, 1.0],
+                    [1.0, 1.0, 3.0, 1.0, 3.0, 9.0],
+                    [1.0, 3.0, 1.0, 9.0, 3.0, 1.0],
+                    [1.0, 1.0, 5.0, 1.0, 5.0, 25.0],
+                    [1.0, 3.0, 3.0, 9.0, 9.0, 9.0],
+                    [1.0, 5.0, 1.0, 25.0, 5.0, 1.0],
+                ]
+            ),
+        ),
+        (
+            test_circuit2,
+            2,
+            np.array(
+                [
+                    [1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0],
+                    [1.0, 1.0, 1.0, 3.0, 1.0, 1.0, 1.0, 3.0, 3.0, 9.0],
+                    [1.0, 1.0, 3.0, 1.0, 1.0, 3.0, 9.0, 1.0, 3.0, 1.0],
+                    [1.0, 3.0, 1.0, 1.0, 9.0, 3.0, 1.0, 3.0, 1.0, 1.0],
+                    [1.0, 1.0, 1.0, 5.0, 1.0, 1.0, 1.0, 5.0, 5.0, 25.0],
+                    [1.0, 1.0, 3.0, 3.0, 1.0, 3.0, 9.0, 3.0, 9.0, 9.0],
+                    [1.0, 3.0, 1.0, 3.0, 9.0, 3.0, 1.0, 9.0, 3.0, 9.0],
+                    [1.0, 1.0, 5.0, 1.0, 1.0, 5.0, 25.0, 1.0, 5.0, 1.0],
+                    [1.0, 3.0, 3.0, 1.0, 9.0, 9.0, 9.0, 3.0, 3.0, 1.0],
+                    [1.0, 5.0, 1.0, 1.0, 25.0, 5.0, 1.0, 5.0, 1.0, 1.0],
+                ]
+            ),
+        ),
+    ],
+)
+def test_sample_matrix(test_circ, test_degree, expected_matrix):
+    assert (expected_matrix == sample_matrix(test_circ, test_degree, 1)).all()
+
+
+@pytest.mark.parametrize(
+    "test_circ, test_degree, test_fold_multiplier, expected_matrix",
+    [
+        (
+            test_circuit1,
+            2,
+            3,
+            [
+                1.5555555555555578,
+                -0.38888888888888934,
+                -0.38888888888888934,
+                0.09722222222222215,
+                0.027777777777777804,
+                0.09722222222222232,
+            ],
+        ),
+        (
+            test_circuit2,
+            2,
+            2,
+            [
+                2.4062499999999956,
+                -0.6874999999999987,
+                -0.6874999999999987,
+                -0.6874999999999987,
+                0.15624999999999956,
+                0.06249999999999997,
+                0.06249999999999997,
+                0.15624999999999956,
+                0.06249999999999997,
+                0.15624999999999956,
+            ],
+        ),
+    ],
+)
+def test_coeffs(test_circ, test_degree, test_fold_multiplier, expected_matrix):
```

Comment:
Instead of checking equality here, should we consider using something like`np.isclose`?

Reply:
Yes, this is a better option than using `np.round`